packages/ui/src/modal/list/FloorDropdown.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { FC, useMemo } from 'react'
+import { useTokens } from '../../hooks'
+import { Dropdown, DropdownMenuItem } from '../../primitives/Dropdown'
+import { Button, Flex, FormatCryptoCurrency, Text } from '../../primitives'
+import { faEllipsis } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { Currency } from '../../types/Currency'
+
+type FloorDropdownProps = {
+  token?: NonNullable<NonNullable<ReturnType<typeof useTokens>>['data']>[0]
+  currency: Currency
+  defaultCurrency?: Currency
+  setPrice: React.Dispatch<React.SetStateAction<string>>
+  setCurrency: (currency: Currency) => void
+}
+
+const FloorDropdown: FC<FloorDropdownProps> = ({
+  token,
+  currency,
+  defaultCurrency,
+  setPrice,
+  setCurrency,
+}) => {
+  const highestTraitFloor = useMemo(() => {
+    // @TODO: do we want to have a max size of attributes?
```

Comment:
@lunarmayor Any thoughts on what we should set the maximum size of attributes that we will iterate through?

Reply:
Haven't noticed it yet no. Just wanted to be prepared for those collections with thousands of attributes